New Tutorial: Demonstrate DP on DataWarehouse for #300

Koukyosyumei opened the pull request:
## Description

This PR aims to close #300.
Although the label of #300 is "status in progress", it seems that no one is working on this issue. 
This notebook consists of two parts. The first part shows how differential privacy works under the assumption that the attacker already knows almost all of the data except one record. In the second one, the attacker tries to identify one person, and the defender uses differential privacy to avoid this. 

It is the first PR for me to this excellent project, so if I misunderstand something, please correct me!

## Affected Dependencies
no

## How has this been tested?
jupyter lab

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

Commit:
first draft for #300

chinmayshah99 commented on run-clang-format.py:
```diff
@@ -143,7 +144,8 @@ def run_clang_format_diff_wrapper(args, file):
         raise
     except Exception as e:
-        raise UnexpectedError("{}: {}: {}".format(file, e.__class__.__name__, e), e)
+        raise UnexpectedError("{}: {}: {}".format(
```
Can we use f-strings here?

Koukyosyumei replied:
It's not what I intended, and I reset the commit

chinmayshah99 commented on tests/algorithms/test_bounded_mean.py:
```diff
@@ -1,10 +1,10 @@
 
 
-expect_near = lambda expected, actual, tol: (
+def expect_near(expected, actual, tol): return (
```
is there a reason to move from lambda to this new implementation?

Koukyosyumei replied:
It's not what I intended, and I reset the commit

Koukyosyumei commented:
@chinmayshah99 
Thanks for your review! The problems you mentioned were not what I intended. I changed some files to pass mypy tests, but it also couldn't pass ci tests. These errors during ci tests are not what this PR aims to solve, so I reset the commit.  Currently, what I have added are only a notebook and csv files.
